cEP-0032.md

```diff
@@ -0,0 +1,638 @@
+organization. It will help community members to know about their performance
+in-comparison to other members. Following are the new enhancments that will be
+added to coala community website:
+
+1. **Display a webpage for-**
+   1. **organization teams**
+
+        This will show information about all [coala teams](https://github.com/orgs/coala/teams).
+        It will definitely not show what permissions are available to team members
+        if a developer is a part of that team. But, it will definitely provide
+        information like *What does this team aim to do?*, *Members count* etc.
+        Clicking on a **team name** will redirect the user to either on GitHub
+        showing all team members or a new webpage which will show all the team
+        members of this coala team.
+
+   2. **contributors' geolocation in form of a map**
+
+        Showing the *global influence of an organization* around the world of open-
+        source organization can be achieved by showing the contributors' geolocation.
+        It can be shown in the form a world map which shows the geolocaton of all
+        contributors that are members of an organization or have been contributing
+        to the organization.
+
+   3. **mentors**
+
+        It will be good if community website can show all the potential mentors that
+        organizaton has. It will allow other organization to get to know more about
+        them.
+
+2. **Add forms for -**
+   1. adding/updating a calendar event
+   2. adding new Google forms
+   3. adding yourself a GSoC participant
+   4. applying as a mentor in GSoC
+   5. joining the community (*newcomer form*)
+   6. getting promoted to developer(*Developer form*)
+   7. Feedback
+
+      Feedback ​ from our developers or users is necessary for every
+      organization, therefore an addition of feedback form to community website
+      will be helpful.
+
+   *Following Validation checks will be performed while:*
+      1. **Participate as a mentor**
+         * A Developer or Maintainer (​ not Newcomer​ )
+         * GSoC Cleared (in any year)
+         * GCI Cleared (in any year) (optional)
+      2. **Participate as a GSoC participant**
+         * cEP Merged
+         * Project Idea PR merged
+         * At-least one accepting Review ​from any of the maintainers
+         * Issue that gives idea to solve project proposal has a label of
+         “Project/Proposal”
+         * Valid project link to projects website 
+      3. Only **coala-developers** (i.e. members part of coala developers team)
+      will be able to add new google forms and add/update calendar events.
+      4. **Joining the coala community** (*newcomers*)
+         * Hasn't joined GitHub recently (i.e. not created account recently)
+         * verify their email address using an email-verifier-api
+         * has atleast 1 source repositories on GitHub
+         * has done any of the github training exercises etc.
+
+3. **Login/Signup using GitHub**
```

Comment:
@jayvdb, needed your views on if this is required or needed to be done?

Reply:
Please also have a look at [this pr discussions](https://github.com/coala/projects/pull/719) when you get time, we needed your suggestions there too.

Reply:
This would be good.  @Jazzband has this for their website.
The problem will be in getting it hosted, if that is required, so definitely plan on it being an optional feature, and ensure it can be tested on your own fork with a Heroku-type hosting, because we will not deploy it until it is finished and we have time to set up an org instance of the backend.

Reply:
Yeah! we will get problem in getting it hosted because the authentication would work only if its a dynamic website which is currently being served as a static website to users

Reply:
@jayvdb Tried to host community website on [heroku](https://kvgarg-community.herokuapp.com/) but unfortunately get a `KeyError`. It's running completely fine when running heroku locally but on deploying to server its raising an exception

Reply:
> 
> 
> @jayvdb Tried to host community website on [heroku](https://kvgarg-community.herokuapp.com/) but unfortunately get a `KeyError`. It's running completely fine when running heroku locally but on deploying to server its raising an exception

@KVGarg on what file or line are you getting the exception? can you show the full error message the python interpreter gives you

Reply:
`python-interpreter` isn't giving any error because git remotes are configured locally. But they aren't being configured on heroku.
Heroku deployed project link - https://kvgarg-community.herokuapp.com/

Reply:
Based on the debug info, this is what I see:
```
When `name` is None, fetch current repo of current org,
    otherwise, fetch a specified repo of current org.
    """
    # Use environment variable URL when Netlify detected
    if not name and os.environ.get('REPOSITORY_URL'):
        return os.environ.get('URL')
    url = get_remote_url() ...
    owner = url.owner
    path = name if name else url.name
    if hoster is not None:
        deploy_url = 'https://%s.%s.io/%s' % (owner, hoster, path)
    elif url.resource == 'github.com':
```
If the git remotes are not configured on heroku, try calling `git_remote_url(“URL”)` in `git.py` with a URL parameter that points to any other location where the remotes are configured. If there is no location, maybe they need to be configured also.

Reply:
As I can see this is covered for Netlify. Maybe if you change the provider from heroku to Netlify, it can solve the problem immediately.

Reply:
`netlify` is a static site generator, right? will that allow us to host it? Try to deployed django forms on netlify but they were not getting deployed

> As I can see this is covered for Netlify. Maybe if you change the provider from heroku to Netlify, it can solve the problem immediately.

Will try to do this!
> If the git remotes are not configured on heroku, try calling git_remote_url(“URL”) in git.py with a URL parameter that points to any other location where the remotes are configured. If there is no location, maybe they need to be configured also.

Reply:
> 
> 
> As I can see this is covered for Netlify. Maybe if you change the provider from heroku to Netlify, it can solve the problem immediately.

I guess deploying on netlify can be achieved by netlify provided features https://www.netlify.com/docs/authentication-providers/ 😮

Reply:
Yes, I haven’t worked with that, but it seems it’s easy.

Reply:
> As I can see this is covered for Netlify. Maybe if you change the provider from heroku to Netlify, it can solve the problem immediately.

@jayvdb @sks444 @margobra8 
I'm able to resolve this problem by setting two environment variables - REPOSITORY_URL and URL which are kind of necessary for coala Community repo. After adding them, the authentication is working.
Can be checked at: https://kvgarg-community.herokuapp.com/accounts/github/login/ after getting authorized it will return to homepage

Reply:
Will look on netlify OAuth too.

Reply:
This can have benefit if I guess `Session` values can be added while a user gets logged in and use that until user is on the website.